tests/integration/lineage.js

```diff
@@ -0,0 +1,489 @@
+  parent: {
+    _id: report_grandparent._id
+  }
+};
+const report_contact = {
+  _id: 'report_contact',
+  type: 'person',
+  name: 'report_contact_name',
+  parent: {
+    _id: report_parent._id,
+    parent: {
+      _id: report_grandparent._id
+    }
+  },
+  reported_date: '5'
+};
+const report_patient = {
+  _id: 'report_patient',
+  patient_id: '12345',
+  type: 'person',
+  name: 'patient_name',
+  parent: {
+    _id: report_contact._id,
+    parent: {
+      _id: report_parent._id,
+      parent: {
+        _id: report_grandparent._id
+      }
+    }
+  },
+  reported_date: '5'
+};
+const report = {
+  _id: 'report',
+  type: 'data_record',
+  form: 'A',
+  contact: {
+    _id: report_contact._id,
+    parent: {
+      _id: report_parent._id,
+      parent: {
+        _id: report_grandparent._id
+      }
+    }
+  },
+  fields: {
+    patient_id: '12345'
+  }
+};
+const stub_contacts = {
+  _id: 'stub_contacts',
+  type: 'data_record',
+  contact: {
+    _id: 'something',
+    parent: {
+      _id: dummyDoc._id
+    }
+  },
+  form: {}
+};
+const stub_parents = {
+  _id: 'stub_parents',
+  type: 'clinic',
+  contact: { _id: 'something' },
+  parent: {
+    _id: 'something_else',
+    parent: { _id: 'dummyDoc' }
+  }
+};
+const sms_doc = {
+  _id: 'sms_doc',
+  type: 'data_record',
+  from: '123',
+  form: 'D',
+  fields: {},
+  sms_message: {
+    message_id: '76992',
+    message: 'somemessage',
+    from: '123',
+    type: 'sms_message',
+    form: 'D'
+  }
+};
+
+const docs = [
+  circular_area,
+  circular_chw,
+  circular_report,
+  dummyDoc,
+  dummyDoc2,
+  fetch_contact,
+  fetch_noContact,
+  no_contact,
+  no_lineageContact,
+  one_parent,
+  place_parent,
+  place_parentContact,
+  place_grandparent,
+  place_grandparentContact,
+  place_contact,
+  place,
+  report_parent,
+  report_parentContact,
+  report_grandparent,
+  report_grandparentContact,
+  report_contact,
+  report_patient,
+  report,
+  stub_contacts,
+  stub_parents,
+  sms_doc
+];
+const docIds = docs.map(doc => doc._id);
+const deleteDocs = ids => {
+  return db.allDocs({
+    keys: ids,
+    include_docs: true
+  }).then(data => {
+    const docs = data.rows.map(row => {
+      const doc = row.doc;
+      if (doc) {
+        doc._deleted = true;
+        return doc;
+      }
+    }).filter(doc => doc);
+    return db.bulkDocs(docs);
+  }).then(() => {
+    return db.compact();
+  });
+};
+
+describe('Lineage', function() {
+  let lineage;
+
+  before(function() {
+    lineage = lineageFactory(Promise, db);
+    return db.bulkDocs(docs).then(result => {
+      console.log(result);
+      return db.allDocs();
+    }).then(result => {
+      console.log(result);
+    }).catch(error => {
+      console.log('error');
+      console.log(error);
+    });
+  });
+
+  after(function() {
+    return deleteDocs(docIds);
+  });
+
+  afterEach(function() {
+    sinon.restore();
+  });
+
+  it('fetchLineageById queries db with correct parameters', function() {
```

Comment:
As discussed, I don't think it's useful to test interactions - prefer testing the output of `lineage.fetchLineageById()`.